probcomp/bayesrest
# Separate command from entrypoint

## What does this do?
Currently the `ENTRYPOINT` directive in our `Dockerfile` looks like this, and we have no `CMD` directive:

```
ENTRYPOINT ["docker-entrypoint.sh", "start.sh", "python", "bayesapi"]
```

This pull request changes our `Dockerfile` so that the `ENTRYPOINT` and `CMD` directives look like this:

```
ENTRYPOINT ["docker-entrypoint.sh", "start.sh"]
CMD ["python", "bayesapi"]
```

## Why should we do this?
When you run a command with Docker the entrypoint and the command essentially get concatenated together, like so: 

```
<entrypoint> <command>
```

`docker-entrypoint.sh` is responsible for activating the Python environment where the probcomp libraries are installed, and [`start.sh`](https://github.com/jupyter/docker-stacks/blob/master/base-notebook/start.sh) does a bunch of stuff to make [jupyter/minimal-notebook](https://github.com/jupyter/docker-stacks/tree/master/minimal-notebook), a Docker image we inherit from, happy. The behavior we want, and what this pull requests achieves, is to prefix all commands that we run in the container with `docker-entrypoint.sh start.sh`.

To ground this in an example: Before this change if you attempted to run a command with `docker run` the full command `docker` would attempt to run would be:

```
docker-entrypoint.sh start.sh python bayesapi <command>
```

In other words it would attempt to run the script `bayesapi` with the _argument_ `<command>`. 

After this change `docker` would instead run `docker-entrypoint.sh start.sh <command>`, which is what we want.

## How do I test this?
Use `docker run` to run the `ipython` command inside a container spawned from our image and verify that the right Python environment has been activated by trying to import a probcomp library:

Start by building the image and then starting it with the command `ipython`:

```
docker build .
…
Successfully built <image-id>
docker run <image-id> ipython
```

This should drop you at the `ipython` prompt. From there try to import `bayeslite`, which should succeed.

```
import bayeslite
```
